api/src/main/java/org/openmrs/module/commonreports/reports/DRCHIVCaScreeningReportManager.java

```diff
@@ -0,0 +1,192 @@
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		
+		// hivCaScreening Grouping
+		CohortCrossTabDataSetDefinition hivCaScreening = new CohortCrossTabDataSetDefinition();
+		hivCaScreening.addParameters(getParameters());
+		rd.addDataSetDefinition(getHivCaScreeningName(), Mapped.mapStraightThrough(hivCaScreening));
+		
+		Map<String, Object> parameterMappings = new HashMap<String, Object>();
+		parameterMappings.put("onOrAfter", "${startDate}");
+		parameterMappings.put("onOrBefore", "${endDate}");
+		parameterMappings.put("startedOnOrAfter", "${startDate}");
+		parameterMappings.put("startedOnOrBefore", "${endDate}");
+		
+		VisitCohortDefinition visits = new VisitCohortDefinition();
+		visits.setVisitTypeList(vs.getAllVisitTypes(false));
+		visits.addParameter(new Parameter("startedOnOrAfter", "On Or After", Date.class));
+		visits.addParameter(new Parameter("startedOnOrBefore", "On Or Before", Date.class));
+		
+		ConceptService cs = Context.getConceptService();
+		
+		// Cervical cancer screening performed
+		CodedObsCohortDefinition caScreening = new CodedObsCohortDefinition();
+		caScreening.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		caScreening.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		caScreening.setOperator(SetComparator.IN);
+		caScreening.setQuestion(cs.getConceptByUuid("e5e99fc7-ff2d-4306-aefd-b87a07fc9ab4")); // Screened for cervical cancer during this visit
+		caScreening.setTimeModifier(TimeModifier.LAST);
+		List<Concept> caScreeningAnswers = new ArrayList<Concept>();
+		caScreeningAnswers.add(cs.getConceptByUuid("165619AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")); // Cervical cancer screening performed
+		caScreening.setValueList(caScreeningAnswers);
+		
+		SqlCohortDefinition ptLivingWithHIVsqd = new SqlCohortDefinition();
+		
+		//What do you want to do? ---> Enroll new Pt in HIV Care
+		String sql = "SELECT DISTINCT p.patient_id FROM patient p WHERE p.voided = 0 "
+		        + "AND EXISTS (SELECT 1 FROM obs o JOIN concept c_question ON o.concept_id = c_question.concept_id JOIN concept c_answer ON o.value_coded = c_answer.concept_id WHERE o.person_id = p.patient_id AND c_question.uuid = '83e40f2c-c316-43e6-a12e-20a338100281' AND c_answer.uuid = '164144AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' AND o.voided = 0);";
+		ptLivingWithHIVsqd.setQuery(sql);
+		
+		// Alive patients
+		BirthAndDeathCohortDefinition livePatients = new BirthAndDeathCohortDefinition();
+		livePatients.setDied(false);
+		
+		CompositionCohortDefinition ccd = new CompositionCohortDefinition();
+		ccd.initializeFromElements(visits, ptLivingWithHIVsqd, caScreening, livePatients);
+		
+		hivCaScreening.addRow(MessageUtil.translate("commonreports.report.drc.caScreening"), ccd, parameterMappings);
+		
+		setColumnNames();
+		
+		GenderCohortDefinition females = new GenderCohortDefinition();
+		females.setFemaleIncluded(true);
+		
+		// 30-49 years
```

Comment:
Is this correct? We only need disaggregation for women between 30 and 49 years old?